pkg/library/status/check.go

```diff
@@ -96,6 +96,11 @@ func CheckPodsState(workspaceID string, namespace string, labelSelector k8sclien
 		if msg, err := CheckPodEvents(&pod, workspaceID, ignoredEvents, clusterAPI); err != nil || msg != "" {
 			return msg, err
 		}
+		if pod.Status.Phase == corev1.PodPending {
```

Comment:
Checking the pod phase could be put inside the `checkPodConditions()` function, or omitted entirely. Not really sure about this one, though in my observation of the original bug, the pod status would always be stuck in the pending phase, so it made sense to add a check for it.